cli/src/commands/interdiff.rs

```diff
@@ -27,25 +27,49 @@ use crate::complete;
 use crate::diff_util::DiffFormatArgs;
 use crate::ui::Ui;
 
-/// Compare the changes of two commits
+/// Show differences between the diffs of two revisions
 ///
-/// This excludes changes from other commits by temporarily rebasing `--from`
-/// onto `--to`'s parents. If you wish to compare the same change across
-/// versions, consider `jj evolog -p` instead.
+/// This is like running `jj diff -r` on each change, then comparing those
+/// results. It answers: "How do the modifications introduced by revision A
+/// differ from the modifications introduced by revision B?"
+///
+/// For example, if two changes both add a feature but implement it
+/// differently, `jj interdiff --from @- --to other` shows what one
+/// implementation adds or removes that the other doesn't.
+///
+/// A common use of this command is to compare how a change has changed
+/// since the last push to a remote:
+///
+/// ```sh
+/// $ jj interdiff --from push-xyz@origin --to push-xyz
+/// ```
+///
+/// This command is different from `jj diff --from A --to B`, which compares
+/// file contents directly. `interdiff` compares what each change does to its
+/// parent in terms of their patches, not their trees. This makes a difference
```

Comment:
nit: I don't think we use "tree" anywhere, at least not in this sense. Use "file contents" instead as before?

Reply:
Yeah, this is a bit too git-brained heh